orca-bakery/src/test/groovy/com/netflix/spinnaker/orca/bakery/pipeline/BakeStageSpec.groovy

```diff
@@ -154,6 +154,51 @@ class BakeStageSpec extends Specification {
     thrown(ConstraintViolationException)
   }
 
+  def "should NOT fail if image names from unrelated bake stages don't match"() {
+    given:
+    def pipeline = pipeline {
+      stage {
+        id = "1"
+        type = "bake"
+        context = [
+          "region": "us-east-1",
+          "regions": ["us-east-1", "us-west-2"]
+        ]
+        status = ExecutionStatus.RUNNING
+      }
+      // this is a sibling bake stage whose child bake contexts should not be included in stage 1's outputs, but are
+      stage {
+        id = "2"
+        type = "bake"
+        context = [
+          "region": "us-east-1",
+          "regions": ["eu-east-1"]
```

Comment:
```suggestion
          "regions": ["us-east-1"]
```

seems odd that this wouldn't be a superset of the single region

Reply:
I know, right? But based on what I saw, only the `regions` field is actually used? I don't know what the singular `region` is used for, if anything...

Reply:
@cfieber Looks like that singular `region` field is not set by the UI, even when baking in a single region:
```json
    {
      "amiSuffix": "${new java.text.SimpleDateFormat(\"yyyyMMddHHmmss\").format(new java.util.Date())}-${baseLabel}",
      "baseLabel": "candidate",
      "baseOs": "xenial",
      "cloudProviderType": "aws",
      "completeOtherBranchesThenFail": true,
      "continuePipeline": false,
      "extendedAttributes": {},
      "failPipeline": false,
      "name": "Bake Xenial Candidate",
      "package": "jnett-dummyapp-server",
      "refId": "1",
      "regions": [
        "us-east-1"
      ],
      "requisiteStageRefIds": [],
      "storeType": "ebs",
      "type": "bake",
      "user": "jnett@netflix.com",
      "vmType": "hvm"
    }
```
Wonder if we should remove it from the tests...